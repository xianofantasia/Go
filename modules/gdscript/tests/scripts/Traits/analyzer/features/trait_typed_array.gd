trait SomeTrait:
	pass

class SomeClass:
	uses SomeTrait

class OtherClass:
	pass

func test():
	var trait_classes: Array[SomeTrait] = [SomeClass.new()]
	trait_classes.append(SomeClass.new())
	trait_classes.append(OtherClass.new())
	var count = 1
	for trait_class in trait_classes:
		print(count)
		count += 1
	print("ok")
