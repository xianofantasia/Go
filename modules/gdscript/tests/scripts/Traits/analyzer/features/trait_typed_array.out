GDTEST_RUNTIME_ERROR
>> ERROR: Condition "!other_script->inherits_script(script)" is true. Returning: false
>>   Attempted to push_back an object into a TypedArray, that does not inherit from 'GDScript'.
>> ERROR: Condition "!_p->typed.validate(value, "push_back")" is true.
1
2
ok
